systemtests/tests/virtualfull-deletedfiles/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+#   BAREOS® - Backup Archiving REcovery Open Sourced
+#
+#   Copyright (C) 2022-2022 Bareos GmbH & Co. KG
+#
+#   This program is Free Software; you can redistribute it and/or
+#   modify it under the terms of version three of the GNU Affero General Public
+#   License as published by the Free Software Foundation and included
+#   in the file LICENSE.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+#   Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+#   02110-1301, USA.
+
+get_filename_component(BASENAME ${CMAKE_CURRENT_BINARY_DIR} NAME)
+create_systemtest(${SYSTEMTEST_PREFIX} ${BASENAME})
+
+set_tests_properties(system:virtualfull-deletedfiles:01-full-backup PROPERTIES COST 100)
+set_tests_properties(system:virtualfull-deletedfiles:02-incremental-backup PROPERTIES DEPENDS "system:virtualfull-deletedfiles:01-full-backup")
+set_tests_properties(system:virtualfull-deletedfiles:03-incremental-backup PROPERTIES DEPENDS "system:virtualfull-deletedfiles:02-incremental-backup")
+set_tests_properties(system:virtualfull-deletedfiles:04-incremental-backup PROPERTIES DEPENDS "system:virtualfull-deletedfiles:03-incremental-backup")
+set_tests_properties(system:virtualfull-deletedfiles:05-consolidate PROPERTIES DEPENDS "system:virtualfull-deletedfiles:04-incremental-backup")
```

Comment:
you're definitely right that a simple "depends" is not sufficient. It would really help  if CTest had something in between fixture and depends.

My reasoning behind splitting this into as many tests as possible was to be able to actually see which part of the scenario fails without the need of digging deep into the whole thing. That is, of course, open to discussion - I'll create an agenda item for next Thursday.

Reply:
I introduced a change in the cmake stuff that provides a simple way to define hard dependencies via fixtures.

Reply:
I guess its fine to leave it as is.  My issue was that it will be more work to recreate the setup if the test fails, but this can be remedied by just overwriting the cleanup and letting ctest do the setup for me.

A second issue was that if one of the daemons crashes during one of the setup tests (for whatever reason) every test after it will timeout (which takes 5min) but i think this is fixed by the `systemtest_requires()` stuff anyways.